include/seahorn/PredicateAbstraction.hh

```diff
@@ -14,6 +14,45 @@
 namespace seahorn
 {
 	using namespace llvm;
+
+	class HornDbModel
```

Comment:
`HornDbModel` should not depend on `ZFixedPoint`. We might want to use it with other solvers. 
There should be a separate method to convert `ZFixedPoint` answer to a `HornDbModel`. 

I'd like to preserve the old API. To query the model, the use provides an `fapp` and gets back a definition for the predicate based on `fapp` arguments. You can use bound-variables `bind::bvar()` to store the solution internally with `bvar(0)` being the first argument, etc.